src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/FindTransactionByIDTest.java

```diff
@@ -0,0 +1,77 @@
+public class FindTransactionByIDTest extends BaseIntegrationTestIT {
+
+    @Autowired
+    FinancialTransactionRepository financialTransactionRepository;
+
+    @Autowired
+    WalletRepository walletRepository;
+
+    @BeforeEach
+    void clearDatabase() {
+        financialTransactionRepository.deleteAll();
+        walletRepository.deleteAll();
+    }
+
+    @DisplayName("Should return proper financial transaction when search Id exist in database")
+    @Test
+    public void testGetFinancialTransactionById_whenFindingTransactionWithExistingId_thenReturnFinancialTransactionWithCorrespondingId() throws Exception {
+        Wallet wallet = walletRepository.save(new Wallet("TestWallet"));
+        FinancialTransaction testFinancialTransaction = createTestFinancialTransaction(wallet, "Test1");
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/transactions/{id}", testFinancialTransaction.getId())
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.id").value(testFinancialTransaction.getId()))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.amount").value(testFinancialTransaction.getAmount()))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.description").value(testFinancialTransaction.getDescription()))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.type").value(testFinancialTransaction.getType().toString()));
+
+        Assertions.assertEquals(1, financialTransactionRepository.count());
+    }
+
+    @DisplayName("Should return status NOT_FOUND when search Id does not exist in database")
+    @Test
+    public void testGetFinancialTransactionById_whenSearchIdDoesNotExistInDatabase_thenReturnErrorNotFound() throws Exception {
+        Wallet wallet = walletRepository.save(new Wallet("TestWallet"));
+        FinancialTransaction testFinancialTransaction = createTestFinancialTransaction(wallet, "Test1");
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/transactions/{id}", 999)
+                        .accept(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.status().isNotFound())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.status").value(ErrorCode.FT001.getBusinessStatus()))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.message").value(ErrorCode.FT001.getBusinessMessage()))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.statusCode").value(ErrorCode.FT001.getBusinessStatusCode()));
+
+        Assertions.assertEquals(1, financialTransactionRepository.count());
+    }
+
+    private FinancialTransaction createTestFinancialTransaction(Wallet wallet, String description) {
+        return financialTransactionRepository.save(FinancialTransaction.builder()
+                .wallet(wallet)
+                .amount(new BigDecimal("5.0"))
+                .date(Instant.ofEpochMilli(0L))
```

Comment:
```suggestion
                .date(Instant.ofEpochMilli(1L))
```
The minimum timestamp value is 1 second.